### PR TITLE
Initial SKR Mini E3 1.0/1.2 Configs

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

Adds configs for BigTreeTech [SKR Mini E3 1.0 & 1.2 boards](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3) based off the Creality Ender-3 config.

### Benefits

Easier upgrade path for users & always up to date configs.

Edit: Don't forget...you likely have twice the free space:

https://github.com/MarlinFirmware/Marlin/blob/865055e07a5de40fa615414eba0dbf0ca05fcbce/platformio.ini#L294-L301

### CODE REVIEW

File: config/examples/BigTreeTech/SKR Mini E3 1.0/Configuration.h [hunk 1]
```diff
@@ -0,0 +1,2198 @@
...
+ *  Reversed Menu Navigation only?    Enable REVERSE_MENU_DIRECTION.
+ *  Reversed Value Editing only?      Enable BOTH options.
+ */
+
+//
+// This option reverses the encoder direction everywhere.
+//
+//  Set this option if CLOCKWISE causes values to DECREASE
+//
+//#define REVERSE_
```
Comment (sjasonsmith) [inline config/examples/BigTreeTech/SKR Mini E3 1.0/Configuration.h]: I don't think you need to use Soft PWM with these. They did it in the BTT fork to work around conflicting timers, which should be fixed properly here.

Comment (thisiskeithb) [reply config/examples/BigTreeTech/SKR Mini E3 1.0/Configuration.h]: I've updated the configs. Thanks!

**Comment by damvcoool** [comment]
Hello I just have a few questions about the configuration of V1.2

- Was the option DISABLE_INACTIVE_EXTRUDER enabled on purpose?
- I see that you enabled MESH_BED_LEVELING for the SKR Mini E3 v1.2 but LCD_BED_LEVELING is not enabled, wouldn't it be needed to do a manual Bed level, and level the corners?
- Was the option NOZZLE_PARK_FEATURE enabled on purpose?
- Was the option NOZZLE_CLEAN_FEATURE enabled on purpose?
- The choice to enable SDSORT_USES_RAM and SDSORT_CACHE_NAMES is it becuase the SKR Mini E3 v1.2 has more ram than the Melzi board?

...and Some comments.
I liked that you enabled LIN_ADVANCE, S_CURVE_ACCELERATION and SQUARE_WAVE_STEPPING by default and disabled ARC_SUPPORT (Specially since CURA does not create GCODE with this)

I have never used USE_BIG_EDIT_FONT, but i will try it, not sure if it's needed on the Ender Default screen tho.

**Comment by thisiskeithb** [comment]
Thanks for the feedback!

> Was the option DISABLE_INACTIVE_EXTRUDER enabled on purpose?

This was done in one of BigTreeTech's configs, but I've reverted the change.

> I see that you enabled MESH_BED_LEVELING for the SKR Mini E3 v1.2 but LCD_BED_LEVELING is not enabled, wouldn't it be needed to do a manual Bed level, and level the corners?

I've pushed a commit to enable `LCD_BED_LEVELING`.

> Was the option NOZZLE_PARK_FEATURE enabled on purpose?
> Was the option NOZZLE_CLEAN_FEATURE enabled on purpose?

These were left enabled on accident while I was setting up `ADVANCED_PAUSE_FEATURE` (just uncomment to enable) and I've pushed a commit that disables them.

> The choice to enable SDSORT_USES_RAM and SDSORT_CACHE_NAMES is it becuase the SKR Mini E3 v1.2 has more ram than the Melzi board?

Yes.

> I have never used USE_BIG_EDIT_FONT, but i will try it, not sure if it's needed on the Ender Default screen tho.

This ~is~ was used in the default Ender-3 config, but I guess it's not now. I've disabled it.

**Comment by reloxx13** [comment]
just want to share my pid bed settings ( for 60°C ), in the example they are rly diffrent.
example has no temp info, it just says "FIND YOUR OWN" with 90°C
hotend pid looks good.

https://github.com/thisiskeithb/Marlin/blob/436782b7f0de680e849b781946d997f947644a37/config/examples/BigTreeTech/SKR%20Mini%20E3%201.0/Configuration.h#L511-L528

```
  //reloxx13 ender 3 skr mini e3 v1.0 M303 E-1 S60 C10 U1
  #define DEFAULT_bedKp 52.40
  #define DEFAULT_bedKi 9.47
  #define DEFAULT_bedKd 193.24
```

```
example
  //120V 250W silicone heater into 4mm borosilicate (MendelMax 1.5+)
  //from FOPDT model - kp=.39 Tp=405 Tdead=66, Tc set to 79.2, aggressive factor of .15 (vs .1, 1, 10)
  #define DEFAULT_bedKp 10.00
  #define DEFAULT_bedKi .023
  #define DEFAULT_bedKd 305.4
```

**Pull request merged**